powerspikegg/go_gateway/gateway.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"fmt"
+	"google.golang.org/grpc"
+	"net/http"
+	"os"
+
+	"net"
+	fetcherpb "powerspike.gg/powerspikegg/rawdata/fetcher/service_gopb"
+	computation "powerspike.gg/powerspikegg/serving/match_computation_gopb"
+)
+
+type gatewayServer struct {
+	matchFetcherClient fetcherpb.MatchFetcherClient
+	computationClient  computation.MatchComputationClient
+	server             *http.Server
+}
+
+func createServer(matchFetcherClient fetcherpb.MatchFetcherClient, computationClient computation.MatchComputationClient) *gatewayServer {
+	mux := http.NewServeMux()
+
+	s := &http.Server{
+		Handler: mux,
+	}
+
+	gs := &gatewayServer{
+		server:             s,
+		matchFetcherClient: matchFetcherClient,
+		computationClient:  computationClient,
+	}
+
+	// /api/summoner/{summonerName}/{region}
+	mux.HandleFunc("/api/summoner/", gs.summonerHandler)
+	// /api/match/{matchID}/{region}
+	mux.HandleFunc("/api/match/", gs.matchHandler)
+	// /api/computation/{matchID}/{summonerID}/{region}
+	mux.HandleFunc("/api/computation/", gs.computationHandler)
+
+	return gs
+}
+
+func main() {
+	conn, err := grpc.Dial("localhost:50001", grpc.WithInsecure())
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
```

Comment:
You may prefix the `err` with some text (such as `unable to reach fetcher server`) because `err` does not contain the context of what this part of the code is doing.